zulipterminal/ui_tools/views.py

```diff
@@ -376,16 +376,17 @@ def __init__(self, view: Any, model: Any,
     def get_next_unread_topic(self) -> Optional[Tuple[int, str]]:
         topics = list(self.model.unread_counts['unread_topics'].keys())
         next_topic = False
-        for topic in topics:
-            if next_topic is True:
+        if self.last_unread_topic not in topics:
+            next_topic = True
+        # loop over topics list twice
+        # for the case that last_unread_topic was
+        # the last valid unread_topic in topics list.
+        for topic in topics*2:
```

Comment:
The `topics*2` is required due to the muted check and removal of the code at the end?

Reply:
It's more complicated than that. Actually, we want to make sure user gets to the next unread topic in case the last_unread_topic was the last unread_topic in the topics list.